crates/bioauth-flow-rpc/src/lib.rs

```diff
@@ -366,17 +367,79 @@ where
             )
             .map_err(AuthenticateError::RuntimeApi).map_err(errtype)?;
 
-        self.pool
+        info!("Bioauth flow - submitting authenticate transaction");
+
+        let tx_hash = self.pool.hash_of(&ext);
+
+        let mut watch = self.pool
             .submit_and_watch(
                 &sp_api::BlockId::Hash(at),
                 sp_runtime::transaction_validity::TransactionSource::Local,
                 ext,
             )
             .await
-            .map_err(AuthenticateError::BioauthTx).map_err(errtype)?;
-
-        info!("Bioauth flow - authenticate transaction complete");
-
-        Ok(())
+            .map_err(AuthenticateError::BioauthTx).map_err(errtype)?.fuse();
+
+        tokio::spawn(async move {
+            loop {
+                if let Some(tx_status) = watch.next().await {
```

Comment:
@dmitrylavrenov this is bad. Tell me why.

Reply:
My stupid bad - fixed https://github.com/humanode-network/humanode/pull/1018/commits/7a509d6ac5ac2008d7aef328efb2e4c7fb5616a8.

Reply:
Much better!